arelle/plugin/validate/NL/rules/nl_kvk.py

```diff
@@ -1120,6 +1123,47 @@ def rule_nl_kvk_4_1_5_2(
                   'Invalid filenames: %(invalidBasenames)s'))
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NL_INLINE_2024
+    ],
+)
+def rule_nl_kvk_4_4_3_1(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    NL-KVK.4.4.3.1: The extension taxonomy MUST not modify (prohibit and/or override) default members assigned to dimensions by the KVK taxonomy
+    """
+    for modelLink in cast(list[ModelLink], val.modelXbrl.baseSets[XbrlConst.dimensionDefault, None, None, None]):
+            if pluginData.isExtensionUri(modelLink.modelDocument.uri, val.modelXbrl):
+                for linkChild in modelLink:
+                    if (isinstance(linkChild,(ModelObject,PrototypeObject)) and
+                            linkChild.get("{http://www.w3.org/1999/xlink}type") == "arc" and
+                            linkChild.get("{http://www.w3.org/1999/xlink}arcrole") == XbrlConst.dimensionDefault):
+                        fromLabel = linkChild.get("{http://www.w3.org/1999/xlink}from")
+                        for fromResource in modelLink.labeledResources[fromLabel]:
+                            if not pluginData.isExtensionUri(fromResource.modelDocument.uri, val.modelXbrl):
+                                yield Validation.error(
+                                codes='NL.NL-KVK.4.4.3.1.extensionTaxonomyOverridesDefaultMembers',
+                                msg=_('A default member does not match the default member settings of the taxonomy.  Update the default member to taxonomy defaults.'),
+                                modelObject=val.modelXbrl.modelDocument
+                                )
+    docsInDTS = pluginData.getDocumentsInDts(val.modelXbrl)
+    for doc in docsInDTS:
+        if doc.type == ModelDocument.Type.LINKBASE:
+            for arcElt in doc.xmlRootElement.iterdescendants(tag="{http://www.xbrl.org/2003/linkbase}definitionArc"):
+                if arcElt.get("use") == "prohibited" and arcElt.get("{http://www.w3.org/1999/xlink}arcrole")  == XbrlConst.dimensionDefault:
+                    yield Validation.error(
+                        codes='NL.NL-KVK.4.4.3.1.extensionTaxonomyOverridesDefaultMembers',
+                        msg=_('A default member is forbidden in the extension taxonomy.  Update the default member to taxonomy defaults.'),
+                        modelObject=val.modelXbrl.modelDocument
+                    )
```

Comment:
With my branch merged, I think you can take advantage of `getExtensionData` and `iterArcsByType`, as well as some constants I added to `XbrlConst`. Other `http://www.w3.org/1999/xlink` QNames used above could probably be added as constants as well.